Session/Home/App Review/AppReviewPromptModel.swift

```diff
@@ -22,30 +22,38 @@ extension AppReviewPromptModel {
     
     /// Determines the initial state of the app review prompt.
     static func loadInitialAppReviewPromptState(using dependencies: Dependencies) -> AppReviewPromptState?  {
-        /// Check if incomplete app review can be shown again to user on next app launch
-        let retryCount = dependencies[defaults: .standard, key: .rateAppRetryAttemptCount]
-        
         var promptState: AppReviewPromptState?
         
-        // A buffer of 24 hours
-        let buffer: TimeInterval = 24 * 60 * 60
+        if shouldShowAppReviewAgain(using: dependencies) {
+            promptState = .rateSession
+            
+        } else if dependencies[defaults: .standard, key: .didShowAppReviewPrompt] && !dependencies[defaults: .standard, key: .didActionAppReviewPrompt] {
+            dependencies[defaults: .standard, key: .didShowAppReviewPrompt] = false
+            
+            promptState = .enjoyingSession
+        }
+        
+        return promptState
+    }
+    
+    static func shouldShowAppReviewAgain(using dependencies: Dependencies) -> Bool {
```

Comment:
This name is misleading - it suggests that it's going to return a flag indicating whether the prompt should be shown but instead it's updating a bunch of values in `userDefaults`, maybe rename it to something like `checkAndRefreshAppReviewState`